dlt/destinations/impl/lancedb/utils.py

```diff
@@ -0,0 +1,53 @@
+import os
+import uuid
+from typing import Sequence, Union, Dict
+
+from dlt.common.schema import TTableSchema
+from dlt.common.schema.utils import get_columns_names_with_prop
+from dlt.common.typing import DictStrAny
+from dlt.destinations.impl.lancedb.configuration import TEmbeddingProvider
+
+
+PROVIDER_ENVIRONMENT_VARIABLES_MAP: Dict[TEmbeddingProvider, str] = {
+    "cohere": "COHERE_API_KEY",
+    "gemini-text": "GOOGLE_API_KEY",
+    "openai": "OPENAI_API_KEY",
+    "huggingface": "HUGGINGFACE_API_KEY",
+}
+
+
+def generate_uuid(data: DictStrAny, unique_identifiers: Sequence[str], table_name: str) -> str:
+    """Generates deterministic UUID - used for deduplication.
+
+    Args:
+        data (Dict[str, Any]): Arbitrary data to generate UUID for.
+        unique_identifiers (Sequence[str]): A list of unique identifiers.
+        table_name (str): LanceDB table name.
+
+    Returns:
+        str: A string representation of the generated UUID.
+    """
+    data_id = "_".join(str(data[key]) for key in unique_identifiers)
+    return str(uuid.uuid5(uuid.NAMESPACE_DNS, table_name + data_id))
+
+
+def list_unique_identifiers(table_schema: TTableSchema) -> Sequence[str]:
+    """Returns a list of unique identifiers for a table.
+
+    Args:
+        table_schema (TTableSchema): a dlt table schema.
+
+    Returns:
+        Sequence[str]: A list of unique column identifiers.
+    """
+    if table_schema.get("write_disposition") == "merge":
```

Comment:
i think if merge keys are defined this should be computed on the merge keys. This whole code is meant for merging right?